packages/integrations/vercel/src/serverless/adapter.ts

```diff
@@ -1,6 +1,7 @@
 import type { AstroAdapter, AstroConfig, AstroIntegration } from 'astro';
 
 import glob from 'fast-glob';
+import { readFileSync } from 'node:fs';
```

Comment:
We would want this script injection available in the `edge` adapter as well as the `serverless` one, I think. Otherwise documenting differences gets a bit tricky. Does that mean we can’t use `node:fs`? (@JuanM04 or @matthewp might know)

I think the better approach might be:

1. Move `analytics.js` to `src/analytics.ts`
2. Expose it in the `package.json` `"exports"` section:
    ```json
    "./analytics": "./dist/analytics.js"
   ```
3. Inject a script that imports it as an npm module:
    ```js
    injectScript('page', 'import "@astrojs/vercel/analytics"');
    ```

Reply:
This would be my suggestion as well just from a maintenance perspective! 

For the record, it's ok to use `fs` here because this is the integration entrypoint, not the code that will actively run on the server.

Reply:
> We would want this script injection available in the `edge` adapter as well as the `serverless` one, I think. Otherwise documenting differences gets a bit tricky. Does that mean we can’t use `node:fs`? (@JuanM04 or @matthewp might know)
> 
> I think the better approach might be:
> 
> 1. Move `analytics.js` to `src/analytics.ts`
> 2. Expose it in the `package.json` `"exports"` section:
>    ```json
>    "./analytics": "./dist/analytics.js"
>    ```
> 3. Inject a script that imports it as an npm module:
>    ```js
>    injectScript('page', 'import "@astrojs/vercel/analytics"');
>    ```

Sure. I'll try this.